### ISSUE TITLE
Getting android.os.strictmode.DiskWriteViolation on client.execute()

### ISSUE THREAD

**Issue by mnaturel** [GoogleCloudPlatform/recaptcha-enterprise-mobile-sdk#99]
**Describe the bug**

When calling `RecaptchaClient.execute()` in an app setup with `StrictMode.ThreadPolicy` we get log warnings on `android.os.strictmode.DiskWriteViolation` and `android.os.strictmode.DiskReadViolation`.

**reCAPTCHA Site Key**

I am not sure this is related to the site key, I may provide it if it is really needed.

**Integration Method**

* [ ] SPM (iOS)
* [ ] Cocoapods (iOS)
* [ ] Direct Download (iOS)
* [x] GMaven (Android)

**SDK Version (e.g. 18.0.1):** 18.4.0

**To Reproduce**

1. Setup app with the given StrictMode policy : 
```kotlin
StrictMode.setThreadPolicy(
    ThreadPolicy.Builder()
        .detectAll()
        .penaltyLog()
        .build(),
)
```
2. On `ViewModel` side, generate token with `LOGIN` action for instance (here `recaptchaClient` is a correctly initialized client) : 
```kotlin
fun test() {
    viewModelScope.launch {
        generateToken()
    }
}

private suspend fun generateToken() = withContext(Dispatchers.IO) {
   recaptchaClient.execute(RecaptchaAction.LOGIN)
}
```
3. On call to the `execute` method see Stacktrace logs such as :
```
StrictMode policy violation; ~duration=5 ms: android.os.strictmode.DiskReadViolation
     at android.os.StrictMode$AndroidBlockGuardPolicy.onReadFromDisk(StrictMode.java:1596)
     at libcore.io.BlockGuardOs.stat(BlockGuardOs.java:389)
     at libcore.io.ForwardingOs.stat(ForwardingOs.java:221)
     at android.app.ActivityThread$AndroidOs.stat(ActivityThread.java:7601)
     at android.system.Os.stat(Os.java:657)
     at android.app.SharedPreferencesImpl.writeToFile(SharedPreferencesImpl.java:819)
     at android.app.SharedPreferencesImpl.access$900(SharedPreferencesImpl.java:59)
     at android.app.SharedPreferencesImpl$2.run(SharedPreferencesImpl.java:672)
     at android.app.SharedPreferencesImpl.enqueueDiskWrite(SharedPreferencesImpl.java:691)
     at android.app.SharedPreferencesImpl.access$100(SharedPreferencesImpl.java:59)
     at android.app.SharedPreferencesImpl$EditorImpl.commit(SharedPreferencesImpl.java:604)
     at com.google.android.recaptcha.internal.zzq.zzb(com.google.android.recaptcha:recaptcha@@18.4.0:4)
     at com.google.android.recaptcha.internal.zzaw.zzl(com.google.android.recaptcha:recaptcha@@18.4.0:8)
     at com.google.android.recaptcha.internal.zzaw.zzh(com.google.android.recaptcha:recaptcha@@18.4.0:1)
     at com.google.android.recaptcha.internal.zzat.invokeSuspend(com.google.android.recaptcha:recaptcha@@18.4.0:5)
     at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
     at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
     at android.os.Handler.handleCallback(Handler.java:938)
     at android.os.Handler.dispatchMessage(Handler.java:99)
     at android.os.Looper.loop(Looper.java:223)
     at android.app.ActivityThread.main(ActivityThread.java:7701)
     at java.lang.reflect.Method.invoke(Native Method)
     at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:952)
```
and
```
StrictMode policy violation; ~duration=4 ms: android.os.strictmode.DiskWriteViolation
     at android.os.StrictMode$AndroidBlockGuardPolicy.onWriteToDisk(StrictMode.java:1552)
     at libcore.io.BlockGuardOs.remove(BlockGuardOs.java:340)
     at libcore.io.ForwardingOs.remove(ForwardingOs.java:185)
     at android.app.ActivityThread$AndroidOs.remove(ActivityThread.java:7619)
     at java.io.UnixFileSystem.delete(UnixFileSystem.java:331)
     at java.io.File.delete(File.java:1037)
     at android.app.SharedPreferencesImpl.writeToFile(SharedPreferencesImpl.java:833)
     at android.app.SharedPreferencesImpl.access$900(SharedPreferencesImpl.java:59)
     at android.app.SharedPreferencesImpl$2.run(SharedPreferencesImpl.java:672)
     at android.app.SharedPreferencesImpl.enqueueDiskWrite(SharedPreferencesImpl.java:691)
     at android.app.SharedPreferencesImpl.access$100(SharedPreferencesImpl.java:59)
     at android.app.SharedPreferencesImpl$EditorImpl.commit(SharedPreferencesImpl.java:604)
     at com.google.android.recaptcha.internal.zzq.zzb(com.google.android.recaptcha:recaptcha@@18.4.0:4)
     at com.google.android.recaptcha.internal.zzaw.zzl(com.google.android.recaptcha:recaptcha@@18.4.0:8)
     at com.google.android.recaptcha.internal.zzaw.zzh(com.google.android.recaptcha:recaptcha@@18.4.0:1)
     at com.google.android.recaptcha.internal.zzat.invokeSuspend(com.google.android.recaptcha:recaptcha@@18.4.0:5)
     at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
     at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
     at android.os.Handler.handleCallback(Handler.java:938)
     at android.os.Handler.dispatchMessage(Handler.java:99)
     at android.os.Looper.loop(Looper.java:223)
     at android.app.ActivityThread.main(ActivityThread.java:7701)
     at java.lang.reflect.Method.invoke(Native Method)
     at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:952)
``` 

**Expected behavior**

We should not see this type of logs when calling `execute` in `Dispatchers.IO` context.

**Screenshots**

N/A

**Xcode version for iOS (please complete the following information):**
 - Version [e.g. Xcode 13.4.1] N/A

**Device (please complete the following information):**

Device 1
 - Model: [e.g. Pixel 7, iPhone6] Pixel 7a
 - OS: [e.g. Android 28, iOS8.1] Android 14.0 (API 34)
 - Emulator/Simulator or real Device? Emulator

Device 2
 - Model: [e.g. Pixel 7, iPhone6] Nokia 7.2
 - OS: [e.g. Android 28, iOS8.1] Android 11.0 (API 30)
 - Emulator/Simulator or real Device? Real device

**Additional context**

I did not see these logs on an Android 6 emulator.

**Comment by walterjgsp** [comment]
Thanks for the report @mnaturel! We will take a look into it.

**Comment by walterjgsp** [comment]
We were able to reproduce @mnaturel and fix from our side. The fix will be in the next release for Android.

**Comment by mnaturel** [comment]
> We were able to reproduce @mnaturel and fix from our side. The fix will be in the next release for Android.

Thank you for the quick analysis and fix!

**Comment by mcorner** [comment]
Aiming for a fix in 18.5.0-beta03 next week

**Comment by walterjgsp** [comment]
Closing this bug as the Android 18.5.0-beta03 was released with a fix for it.
